Tests/Shopware/Connect/Component/CategoryExtractorTest.php

```diff
@@ -650,5 +650,51 @@ public function testGetRemoteCategoriesTreeByStream()
         $remoteCategories = $this->categoryExtractor->getRemoteCategoriesTreeByStream($stream, $shopId);
         $this->assertEquals($expected, $remoteCategories);
     }
+
+    /**
+     * Test concat shopId, categoryId and unique number
+     */
+    public function testConcatShopIdAndCategoryId()
+    {
+        $shopId = 1;
+
+        $randomStringGenerator = $this->getMockBuilder('\\ShopwarePlugins\\Connect\\Components\\RandomStringGenerator')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $categoryExtractor = new \ShopwarePlugins\Connect\Components\CategoryExtractor(
+            $this->attributeRepository,
+            new \ShopwarePlugins\Connect\Components\CategoryResolver\AutoCategoryResolver(
+                $this->em,
+                $this->em->getRepository('Shopware\Models\Category\Category'),
+                $this->em->getRepository('Shopware\CustomModels\Connect\RemoteCategory'),
+                new \ShopwarePlugins\Connect\Components\Config($this->em)
+            ),
+            $this->configurationGateway,
+            $randomStringGenerator
+        );
+
+        $argument = sprintf('shopId%s~%s', $shopId, '/Kleidung-unit/Hosen-unit/Hosentraeger-unit');
+        $randomStringGenerator->expects($this->once())
+            ->method('generate')
+            ->with($argument)
+            ->willReturn($argument . '1040');
+
+        $expected = array(
```

Comment:
I wrote the expected results in own variables too, but meanwhile I prefer to put it directly into the `assert*`-Calls. 

Like:
```
$this->assertEquals([
    [
             'id' => 'shopId1~/Kleidung-unit/Hosen-unit/Hosentraeger-unit1040',
             'categoryId' => '/Kleidung-unit/Hosen-unit/Hosentraeger-unit',
             'name' => 'Hosentraeger',
             'leaf' => true,
             'children' => array(),
              'cls' => "sc-tree-node",
             'expanded' => false
    ],
    $result
]);
```
Most of the time it is more obvious to find the expected result, further you dont need to jump around with the eyes to find the expected value if a test failed.
But the most important point is that it breaks the readability of the test because the setup of all values increases dramatically, especially with bigger arrays.
It's my preferred way, you can do it like this, just a suggestion if you like it :)